rholang/src/main/scala/coop/rchain/rholang/interpreter/accounting/package.scala

```diff
@@ -15,10 +15,9 @@ package object accounting extends Costs {
       amount: Cost
   )(implicit cost: _cost[F], error: _error[F]): F[Unit] =
     for {
-      currentCost <- cost.get
-      newCost     = currentCost - amount
-      _           <- cost.set(newCost)
-      _           <- error.ensure(cost.get)(OutOfPhlogistonsError)(_.value >= 0)
+      _ <- error.ensure(cost.get)(OutOfPhlogistonsError)(_.value >= 0)
```

Comment:
I am not sure that double check is necessary. If every time the phlos available is updated with this method then if the limit goes below 0 it will break during exit.

Reply:
good point! Having just the check immediately after the `modify` should suffice :+1: